src/lib/hooks/useExchangeRates.ts

```diff
@@ -1,26 +1,58 @@
 import { useEffect, useState } from 'react';
+import { getLocalValues, setLocalValue } from '@/lib/utils/localStorage';
 import { general } from '@/lib/data/general';
 
+const ONE_MINUTE_IN_MS = 60000;
+const MAX_COINGECKO_REQUESTS_PER_MINUTE =
+    import.meta.env.VITE_MAX_COINGECKO_REQUESTS_PER_MINUTE ?? 10;
+const COINGECKO_API_URL = 'https://api.coingecko.com/api/v3/simple/price';
+const COIN_ID = 'ark';
+
+const fetchRates = async () => {
+    const url = new URL(COINGECKO_API_URL);
+    url.searchParams.set('ids', COIN_ID);
+    url.searchParams.set(
+        'vs_currencies',
+        general.currencies.map((currency) => currency.value).join(','),
+    );
+
+    const response = await fetch(url);
+    const parsedRates = await response.json();
+    return parsedRates[COIN_ID];
+};
+
 export const useExchangeRates = () => {
     const [rates, setRates] = useState<Record<string, number>>({});
     const [isLoading, setLoading] = useState<boolean>(true);
 
-    const coinId = 'ark';
-    const coingeckoApiUrl = 'https://api.coingecko.com/api/v3/simple/price';
-
     useEffect(() => {
-        const url = new URL(coingeckoApiUrl);
-        url.searchParams.set('ids', coinId);
-        url.searchParams.set(
-            'vs_currencies',
-            general.currencies.map((currency) => currency.value).join(','),
-        );
-
         const getRates = async () => {
-            const rates = await fetch(url);
-            const parsedRates = await rates.json();
+            const now = Date.now();
+            let { ratesCache } = await getLocalValues();
+
+            if (!ratesCache) {
+                ratesCache = { lastFetch: 0, rates: {}, requestCount: 0 };
+            }
+
+            if (now - ratesCache.lastFetch < ONE_MINUTE_IN_MS) {
+                if (ratesCache.requestCount < MAX_COINGECKO_REQUESTS_PER_MINUTE) {
+                    ratesCache.rates = await fetchRates();
+                    ratesCache.requestCount += 1;
+                } else {
+                    // If MAX_COINGECKO_REQUESTS_PER_MINUTE is reached, use cached rates
+                    setRates(ratesCache.rates);
+                    setLoading(false);
+                    return;
+                }
```

Comment:
```javascript
                if (ratesCache.requestCount < MAX_COINGECKO_REQUESTS_PER_MINUTE) {
                    ratesCache.rates = await fetchRates();
                    ratesCache.requestCount += 1;
                } else {
                    // If MAX_COINGECKO_REQUESTS_PER_MINUTE is reached, use cached rates
                    setRates(ratesCache.rates);
                    setLoading(false);
                    return;
                }
```

What do you think about removing this part?  1 min data is fresh enough I think, I wouldn't keep sending 10 requests in a minute - even it is allowed.